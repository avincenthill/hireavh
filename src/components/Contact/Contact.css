.contact {
  margin: calc(var(--s-xs)/2);
  background-color: var(--c-black);
  width: calc(var(--s-xl)*1.1);
  color: var(--c-1);
  display: flex;
  align-items: center;
  justify-content: space-between;
  border-radius: var(--s-xs);
  box-shadow: inset 0 0 0 var(--b-1) var(--c-1);
  box-sizing: border-box;
}

.contact-container {
  margin-bottom: var(--s-l);
}

.contact-title {
  margin-bottom: var(--s-xs);
}

.contact-img {
  margin-top: var(--s-s);
  width: 100px;
  height: 100px;
  border-radius: 100vmax;
  border: var(--c-1) solid var(--b-1);
}

.contact-button {
  background-color: var(--c-4);
  border-radius: 0 var(--s-xs) var(--s-xs) 0;
  border: var(--c-1) solid var(--b-1);
}

.contact-a {
  font-size: var(--fs-xs);
  text-decoration: none;
  color: var(--c-1)
}