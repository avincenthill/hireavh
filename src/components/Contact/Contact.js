import React, { Component } from "react";
import Page from "components/Page/Page";
import content from "content/content";
import styleconfig from "styles/styleconfig";
import { IconContext } from "react-icons";
import {
  GoMail,
  GoDeviceMobile,
  GoFile,
  GoCloudDownload,
} from "react-icons/go";
import { FaLinkedinIn, FaGithub, FaRegCopy, FaFileCode } from "react-icons/fa";
import ClipboardJS from "clipboard";
import headshot from "assets/img/headshot.png";
import "./Contact.css";

export default class Contact extends Component {
  componentDidMount() {
    const classes = [
      ".contact-button-email",
      ".contact-button-cell",
      ".contact-button-linkedin",
      ".contact-button-github",
    ];
    classes.forEach((className) => {
      new ClipboardJS(className);
    });
  }

  render() {
    return (
      <div className="contact-container">
        <Page>
          <IconContext.Provider value={{ style: styleconfig.icons.contact }}>
            {/* headshot */}
            <img className="contact-img" src={headshot} alt="headshot"></img>

            {/* name */}
            <h3 className="contact-title">{content.contact.name}</h3>

            {/* email */}
            <div className="contact">
              <a className="contact-a" href={`mailto:${content.contact.email}`}>
                <GoMail />
              </a>
              <a className="contact-a" href={`mailto:${content.contact.email}`}>
                <span>{content.contact.email}</span>
              </a>
              <button
                data-clipboard-text={content.contact.email}
                className="contact-button contact-button-linkedin"
              >
                <FaRegCopy />
              </button>
            </div>

            {/* cell */}
            <div className="contact">
              <a className="contact-a" href={`tel:+1-${content.contact.cell}`}>
                <GoDeviceMobile />
              </a>
              <a className="contact-a" href={`tel:+1-${content.contact.cell}`}>
                <span>{content.contact.cell}</span>
              </a>
              <button
                data-clipboard-text={content.contact.cell}
                className="contact-button contact-button-linkedin"
              >
                <FaRegCopy />
              </button>
            </div>

            {/* linkedin */}
            <div className="contact">
              <a
                className="contact-a"
                href={`https://${content.contact.linkedin}`}
              >
                <FaLinkedinIn />
              </a>
              <a
                className="contact-a"
                href={`https://${content.contact.linkedin}`}
              >
                <span>{content.contact.linkedin}</span>
              </a>
              <button
                data-clipboard-text={content.contact.linkedin}
                className="contact-button contact-button-linkedin"
              >
                <FaRegCopy />
              </button>
            </div>

            {/* github */}
            <div className="contact">
              <a
                className="contact-a"
                href={`https://${content.contact.github}`}
              >
                <FaGithub />
              </a>
              <a
                className="contact-a"
                href={`https://${content.contact.github}`}
              >
                <span>{content.contact.github}</span>
              </a>
              <button
                data-clipboard-text={content.contact.github}
                className="contact-button contact-button-github"
              >
                <FaRegCopy />
              </button>
            </div>

            {/* site */}
            <div className="contact">
              <a className="contact-a" href={`https://${content.contact.site}`}>
                <FaFileCode />
              </a>
              <a className="contact-a" href={`https://${content.contact.site}`}>
                <span>{content.contact.site}</span>
              </a>
              <button
                data-clipboard-text={content.contact.site}
                className="contact-button contact-button-github"
              >
                <FaRegCopy />
              </button>
            </div>

            {/* resume */}
            <a
              className="contact contact-a"
              href={require("assets/pdf/avh_resume.pdf")}
              download={content.resume.downloadName}
            >
              <GoFile />
              <span>{content.contact.resume}</span>
              <button className="contact-button">
                <GoCloudDownload />
              </button>
            </a>
          </IconContext.Provider>
        </Page>
      </div>
    );
  }
}
