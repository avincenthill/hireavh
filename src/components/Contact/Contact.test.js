import Contact from "./Contact";
import React from "react";
// import { shallow } from "enzyme";
import renderer from "react-test-renderer";

describe("Contact", () => {
  describe("snapshot", () => {
    it("should match the last saved snapshot", () => {
      const tree = renderer.create(<Contact />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
