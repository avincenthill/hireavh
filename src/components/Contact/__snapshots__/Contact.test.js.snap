// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact snapshot should match the last saved snapshot 1`] = `
<div
  className="contact-container"
>
  <div>
    <div>
      <div
        className="nav nav-top"
      >
        <div
          className="nav-icon $
          this.props.isBottom ? \\"nav-icon-bottom\\" : \\"nav-icon-top\\"
        }"
        >
          <svg
            fill="currentColor"
            height="1em"
            onClick={[Function]}
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": "var(--c-1)",
                "height": "3rem",
                "width": "3rem",
              }
            }
            viewBox="0 0 8 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 4v1H0V4h8zM0 8h8V7H0v1zm0 3h8v-1H0v1z"
              fillRule="evenodd"
            />
          </svg>
        </div>
      </div>
      <a
        className="header-container"
        href="/about"
      >
        <div
          className="header-switch-theme"
        >
          <div
            className="toggle-switch"
          >
            <input
              className="toggle-switch-checkbox"
              defaultChecked={true}
              id="toggle-switch-theme"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              className="toggle-switch-label"
              htmlFor="toggle-switch-theme"
            >
              <span
                className="toggle-switch-inner"
                data-no="üåô"
                data-yes="‚òÄÔ∏è"
              />
            </label>
          </div>
        </div>
        üè† about
      </a>
    </div>
    <div
      className="page-container"
    >
      <h3
        className="contact-title"
      >
        üë®‚Äçüíª Alex "AVH" Vincent-Hill
      </h3>
      <div
        className="contact button-hover-light"
      >
        <a
          className="contact-a"
          href="mailto:alexanderjvincenthill@gmail.com"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": "var(--c-1)",
                "height": "var(--s-xs)",
                "padding": ".5rem",
                "width": "var(--s-xs)",
              }
            }
            viewBox="0 0 14 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 4v8c0 .55.45 1 1 1h12c.55 0 1-.45 1-1V4c0-.55-.45-1-1-1H1c-.55 0-1 .45-1 1zm13 0L7 9 1 4h12zM1 5.5l4 3-4 3v-6zM2 12l3.5-3L7 10.5 8.5 9l3.5 3H2zm11-.5l-4-3 4-3v6z"
              fillRule="evenodd"
            />
          </svg>
        </a>
        <a
          className="contact-a"
          href="mailto:alexanderjvincenthill@gmail.com"
        >
          <span>
            alexanderjvincenthill@gmail.com
          </span>
        </a>
        <button
          className="contact-button contact-button-linkedin"
          data-clipboard-text="alexanderjvincenthill@gmail.com"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": "var(--c-1)",
                "height": "var(--s-xs)",
                "padding": ".5rem",
                "width": "var(--s-xs)",
              }
            }
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M433.941 65.941l-51.882-51.882A48 48 0 0 0 348.118 0H176c-26.51 0-48 21.49-48 48v48H48c-26.51 0-48 21.49-48 48v320c0 26.51 21.49 48 48 48h224c26.51 0 48-21.49 48-48v-48h80c26.51 0 48-21.49 48-48V99.882a48 48 0 0 0-14.059-33.941zM266 464H54a6 6 0 0 1-6-6V150a6 6 0 0 1 6-6h74v224c0 26.51 21.49 48 48 48h96v42a6 6 0 0 1-6 6zm128-96H182a6 6 0 0 1-6-6V54a6 6 0 0 1 6-6h106v88c0 13.255 10.745 24 24 24h88v202a6 6 0 0 1-6 6zm6-256h-64V48h9.632c1.591 0 3.117.632 4.243 1.757l48.368 48.368a6 6 0 0 1 1.757 4.243V112z"
            />
          </svg>
        </button>
      </div>
      <div
        className="contact button-hover-light"
      >
        <a
          className="contact-a"
          href="https://linkedin.com/in/avincenthill"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": "var(--c-1)",
                "height": "var(--s-xs)",
                "padding": ".5rem",
                "width": "var(--s-xs)",
              }
            }
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M100.28 448H7.4V148.9h92.88zM53.79 108.1C24.09 108.1 0 83.5 0 53.8a53.79 53.79 0 0 1 107.58 0c0 29.7-24.1 54.3-53.79 54.3zM447.9 448h-92.68V302.4c0-34.7-.7-79.2-48.29-79.2-48.29 0-55.69 37.7-55.69 76.7V448h-92.78V148.9h89.08v40.8h1.3c12.4-23.5 42.69-48.3 87.88-48.3 94 0 111.28 61.9 111.28 142.3V448z"
            />
          </svg>
        </a>
        <a
          className="contact-a"
          href="https://linkedin.com/in/avincenthill"
        >
          <span>
            linkedin.com/in/avincenthill
          </span>
        </a>
        <button
          className="contact-button contact-button-linkedin"
          data-clipboard-text="linkedin.com/in/avincenthill"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": "var(--c-1)",
                "height": "var(--s-xs)",
                "padding": ".5rem",
                "width": "var(--s-xs)",
              }
            }
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M433.941 65.941l-51.882-51.882A48 48 0 0 0 348.118 0H176c-26.51 0-48 21.49-48 48v48H48c-26.51 0-48 21.49-48 48v320c0 26.51 21.49 48 48 48h224c26.51 0 48-21.49 48-48v-48h80c26.51 0 48-21.49 48-48V99.882a48 48 0 0 0-14.059-33.941zM266 464H54a6 6 0 0 1-6-6V150a6 6 0 0 1 6-6h74v224c0 26.51 21.49 48 48 48h96v42a6 6 0 0 1-6 6zm128-96H182a6 6 0 0 1-6-6V54a6 6 0 0 1 6-6h106v88c0 13.255 10.745 24 24 24h88v202a6 6 0 0 1-6 6zm6-256h-64V48h9.632c1.591 0 3.117.632 4.243 1.757l48.368 48.368a6 6 0 0 1 1.757 4.243V112z"
            />
          </svg>
        </button>
      </div>
      <div
        className="contact button-hover-light"
      >
        <a
          className="contact-a"
          href="https://github.com/avincenthill"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": "var(--c-1)",
                "height": "var(--s-xs)",
                "padding": ".5rem",
                "width": "var(--s-xs)",
              }
            }
            viewBox="0 0 496 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
            />
          </svg>
        </a>
        <a
          className="contact-a"
          href="https://github.com/avincenthill"
        >
          <span>
            github.com/avincenthill
          </span>
        </a>
        <button
          className="contact-button contact-button-github"
          data-clipboard-text="github.com/avincenthill"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": "var(--c-1)",
                "height": "var(--s-xs)",
                "padding": ".5rem",
                "width": "var(--s-xs)",
              }
            }
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M433.941 65.941l-51.882-51.882A48 48 0 0 0 348.118 0H176c-26.51 0-48 21.49-48 48v48H48c-26.51 0-48 21.49-48 48v320c0 26.51 21.49 48 48 48h224c26.51 0 48-21.49 48-48v-48h80c26.51 0 48-21.49 48-48V99.882a48 48 0 0 0-14.059-33.941zM266 464H54a6 6 0 0 1-6-6V150a6 6 0 0 1 6-6h74v224c0 26.51 21.49 48 48 48h96v42a6 6 0 0 1-6 6zm128-96H182a6 6 0 0 1-6-6V54a6 6 0 0 1 6-6h106v88c0 13.255 10.745 24 24 24h88v202a6 6 0 0 1-6 6zm6-256h-64V48h9.632c1.591 0 3.117.632 4.243 1.757l48.368 48.368a6 6 0 0 1 1.757 4.243V112z"
            />
          </svg>
        </button>
      </div>
      <a
        className="contact button-hover-light contact-a"
        download="Resume_Alex_AVH_Vincent-Hill.pdf"
        href="avh_resume.pdf"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": "var(--c-1)",
              "height": "var(--s-xs)",
              "padding": ".5rem",
              "width": "var(--s-xs)",
            }
          }
          viewBox="0 0 12 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M6 5H2V4h4v1zM2 8h7V7H2v1zm0 2h7V9H2v1zm0 2h7v-1H2v1zm10-7.5V14c0 .55-.45 1-1 1H1c-.55 0-1-.45-1-1V2c0-.55.45-1 1-1h7.5L12 4.5zM11 5L8 2H1v12h10V5z"
            fillRule="evenodd"
          />
        </svg>
        <span>
          resume
        </span>
        <button
          className="contact-button"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": "var(--c-1)",
                "height": "var(--s-xs)",
                "padding": ".5rem",
                "width": "var(--s-xs)",
              }
            }
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9 12h2l-3 3-3-3h2V7h2v5zm3-8c0-.44-.91-3-4.5-3C5.08 1 3 2.92 3 5 1.02 5 0 6.52 0 8c0 1.53 1 3 3 3h3V9.7H3C1.38 9.7 1.3 8.28 1.3 8c0-.17.05-1.7 1.7-1.7h1.3V5c0-1.39 1.56-2.7 3.2-2.7 2.55 0 3.13 1.55 3.2 1.8v1.2H12c.81 0 2.7.22 2.7 2.2 0 2.09-2.25 2.2-2.7 2.2h-2V11h2c2.08 0 4-1.16 4-3.5C16 5.06 14.08 4 12 4z"
              fillRule="evenodd"
            />
          </svg>
        </button>
      </a>
      <div>
        <br />
        <b>
          Buy me a coffee:
        </b>
        <br />
        <br />
        <form
          action="https://www.paypal.com/donate"
          method="post"
          target="_top"
        >
          <input
            name="cmd"
            type="hidden"
            value="_donations"
          />
          <input
            name="business"
            type="hidden"
            value="alexanderjvincenthill@gmail.com"
          />
          <input
            name="currency_code"
            type="hidden"
            value="USD"
          />
          <input
            alt="Donate with PayPal button"
            border="0"
            name="submit"
            src="https://www.paypalobjects.com/en_US/i/btn/btn_donate_SM.gif"
            title="PayPal - The safer, easier way to pay online!"
            type="image"
          />
          <img
            alt=""
            border="0"
            height="1"
            src="https://www.paypal.com/en_US/i/scr/pixel.gif"
            width="1"
          />
        </form>
      </div>
    </div>
    <div>
      <div
        className="nav nav-bottom"
      >
        <div
          className="nav-icon $
          this.props.isBottom ? \\"nav-icon-bottom\\" : \\"nav-icon-top\\"
        }"
        >
          <svg
            fill="currentColor"
            height="1em"
            onClick={[Function]}
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": "var(--c-1)",
                "height": "3rem",
                "width": "3rem",
              }
            }
            viewBox="0 0 8 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 4v1H0V4h8zM0 8h8V7H0v1zm0 3h8v-1H0v1z"
              fillRule="evenodd"
            />
          </svg>
        </div>
      </div>
      <a
        className="footer-container"
        href="/about"
      >
        <span>
          üè† about
        </span>
      </a>
    </div>
  </div>
</div>
`;
